### DISCUSSION TITLE
Default date format for v3

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#7699] [Help] [answered]
Hello,

I'm unable to find this. In v2 you can define a default date format for forms and tables, but I'm unable to find that for v3. Is this still possible?

**Comment by danharrin** [comment, marked as answer]
Classes like `DateTimePicker`, `Table` and `Infolist` have static properties like `$defaultDateTimeDisplayFormat` which you can set from a service provider:

```
DateTimePicker::$defaultDateTimeDisplayFormat = 'xxx'
```

**Reply by underdpt** [reply]
Thanks!

**Comment by malle-pietje** [comment]
Nice trick! It does require you to call the `->dateTime()` method on the column, though.

**Reply by ya-cha** [reply]
Just in case someone lands here like me: There is also the `$defaultDateDisplayFormat`

**Comment by jpProeasyprint** [comment]
For the record, if you need to make the format dependent of the selected locale, setting directly the static member is too early in the lifecycle; Setting with a closure work (the format string is in the translations file)
`DateTimePicker::configureUsing(fn () => DateTimePicker::$defaultDateDisplayFormat = __('format.date'));
DateTimePicker::configureUsing(fn () => DateTimePicker::$defaultDateTimeDisplayFormat = __('format.dateTime'));`

**Comment by prog47** [comment]
It seems for v4 we need another approach, because now it is `protected` & I get this error:
`Cannot access protected property Filament\Forms\Components\DateTimePicker::$defaultDateDisplayFormat`

**Comment by jpProeasyprint** [comment]
The dateTimePicker is available in the closure : `DateTimePicker::configureUsing(fn (DateTimePicker $dateTimePicker) => $dateTimePicker->defaultDateDisplayFormat(__('format.datePhp')));`
